url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/io/filen/ReferenceModifierKyoroFile.java

```diff
@@ -0,0 +1,120 @@
+package net.hetimatan.io.filen;
+
+
+import java.io.IOException;
+import java.util.LinkedList;
+
+import net.hetimatan.io.file.KyoroFile;
+
+//
+// todo:RACashFileと同じコードがある。 
+// RACashFileの一部をKyoroFileとして取得する。
+// 独自にキャッシュをもてる。 
+public class ReferenceModifierKyoroFile implements KyoroFile {
+	private KyoroFile mRAFile = null;
+	private LinkedList<ByteKyoroFile> mCash = null;
+	private long mFilePointer = 0;
+	private long mStart = 0;
+	private long mEnd = 0;
+	private int mChunkSize = 0;
+	private boolean mIsCash = false;
+
+	public ReferenceModifierKyoroFile(KyoroFile f, long start, long end, int chunkSize, int chunkNum) throws IOException {		
+		mChunkSize = chunkSize;
+		mRAFile = f;
+		mIsCash = false;
+		mStart = start;
+		mEnd = end;
+		mCash = new LinkedList<ByteKyoroFile>();
+		for(int i=0;i<chunkNum;i++) {
+			ByteKyoroFile cash = new ByteKyoroFile(mChunkSize, mChunkSize);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code might contain a resource leak. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *cash*. Currently, there are execution paths that do not contain closure statements, for example, when ByteKyoroFile.skip throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).